cluster-autoscaler/simulator/node_info_utils.go

```diff
@@ -41,6 +42,9 @@ type nodeGroupTemplateNodeInfoGetter interface {
 // SanitizedTemplateNodeInfoFromNodeGroup returns a template NodeInfo object based on NodeGroup.TemplateNodeInfo(). The template is sanitized, and only
 // contains the pods that should appear on a new Node from the same node group (e.g. DaemonSet pods).
 func SanitizedTemplateNodeInfoFromNodeGroup(nodeGroup nodeGroupTemplateNodeInfoGetter, daemonsets []*appsv1.DaemonSet, taintConfig taints.TaintConfig) (*framework.NodeInfo, errors.AutoscalerError) {
+	// TODO(DRA): Figure out how to handle TemplateNodeInfo() returning DaemonSet Pods using DRA. Currently, things only work correctly if such pods are
+	// already allocated by TemplateNodeInfo(). It might be better for TemplateNodeInfo() to return unallocated claims, and to run scheduler predicates and
+	// compute the allocations here.
```

Comment:
https://github.com/kubernetes/autoscaler/pull/7530/#issuecomment-2536807564:
> "Sanitizing" ResourceClaims (4th commit). During the simulations, CA duplicates Nodes and their DaemonSet pods. Duplicated Nodes/Pods go through "sanitization" so that they aren't actually identical to the originals. Doing this for DS pods using DRA has some corner cases (described in the comments), WDYT about this? Do we anticipate DS pods using DRA?

A use case for a DS with claims is for monitoring, in which the claims will use "admin mode", meaning that they don't block allocating the monitored devices for some other pod. Admin mode is under a feature gate in Kubernetes and alpha, so supporting this in CA is not a priority.